### PR TITLE
[Enhancement] Optimize disorder publish request

### PR DESCRIPTION
**Pull request by meegoo**
## What type of PR is this：
- [ ] bug
- [ ] feature
- [x] enhancement
- [ ] others

## Which issues of this PR fixes ：

Fixes #

## Problem Summary(Required) ：

publish request may be disorder since network failure etc
reorder publish request by version, make FE publish success faster, and avoid pending rowsets too much error when using primary key.

### CODE REVIEW

File: be/src/agent/task_worker_pool.cpp [hunk 1]
```diff
@@ -822,91 +822,118 @@ void* TaskWorkerPool::_publish_version_worker_thread_callback(void* arg_this) {
...
+                for (auto& rhs_partition : rhs->publish_version_req.partition_version_infos) {
+                    if (lhs_partition.partition_id == rhs_partition.partition_id &&
+                        lhs_partition.version > rhs_partition.version) {
+                        return true;
+                    }
+                    break;
+                }
+                break;
+            }
+            return false;
```
Comment (decster) [inline be/src/agent/task_worker_pool.cpp:840]: Is this guarantee total order?  If lhs.partition_id != rhs.partition_id the result is always false, which means `lhs > rhs` and `rhs < rhs` can be both false

Comment (meegoo) [reply be/src/agent/task_worker_pool.cpp:840]: just try best, there is an inconsistency between the partitions requested before and after, but the probability is low

Comment (decster) [reply be/src/agent/task_worker_pool.cpp:840]: comparator should provide `strict weak ordering`, or this priority_queue won't work. 
https://en.cppreference.com/w/cpp/named_req/Compare

Comment (decster) [reply be/src/agent/task_worker_pool.cpp]: why not just
`lhs->publish_version_req.partition_version_infos[0].verfsion > rhs->publish_version_req.partition_version_infos[0].verfsion`

Comment (meegoo) [reply be/src/agent/task_worker_pool.cpp]: handle empty partition_version_infos

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 0 / 8 (00.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/task/PublishVersionTask.java|0|2|00.00%|
|:large_blue_circle:|com/starrocks/catalog/TabletInvertedIndex.java|0|2|00.00%|
|:large_blue_circle:|com/starrocks/master/ReportHandler.java|0|4|00.00%|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 0 / 8 (00.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/task/PublishVersionTask.java|0|2|00.00%|
|:large_blue_circle:|com/starrocks/catalog/TabletInvertedIndex.java|0|2|00.00%|
|:large_blue_circle:|com/starrocks/master/ReportHandler.java|0|4|00.00%|

**Review by decster — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/task/PublishVersionTask.java [hunk 2]
```diff
@@ -42,19 +42,22 @@ public class PublishVersionTask extends AgentTask {
     private List<TPartitionVersionInfo> partitionVersionInfos;
     private List<Long> errorTablets;
     private boolean isFinished;
+    private long commitTime;
```
Comment (chaoyli) [inline fe/fe-core/src/main/java/com/starrocks/task/PublishVersionTask.java:45]: commitTimestamp is a better name. You can be unified with the thrift.

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 0 / 8 (00.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/task/PublishVersionTask.java|0|2|00.00%|
|:large_blue_circle:|com/starrocks/catalog/TabletInvertedIndex.java|0|2|00.00%|
|:large_blue_circle:|com/starrocks/master/ReportHandler.java|0|4|00.00%|

**Review by decster — APPROVED** [review]

**Review by chaoyli — APPROVED** [review]

**Comment by meegoo** [comment]
@mergify backport branch-2.1

**Pull request merged**